src/kayenta/edit/filterTemplatesValidation.ts

```diff
@@ -0,0 +1,94 @@
+import { get, isEmpty } from 'lodash';
+
+import { ICanaryMetricConfig } from '../domain';
+import { IEditingTemplateState } from '../reducers/editingTemplate';
+
+export interface ICanaryFilterTemplateValidationMessage {
+  message: string;
+}
+
+export interface ICanaryFilterTemplateValidationMessages {
+  errors: { [key: string]: ICanaryFilterTemplateValidationMessage };
+  warnings: { [key: string]: ICanaryFilterTemplateValidationMessage };
+}
+
+export interface ITemplateValidationInput {
+  editingTemplate: IEditingTemplateState;
+  configTemplates: { [name: string]: string };
+  metricList: ICanaryMetricConfig[];
+  editingMetric: ICanaryMetricConfig;
+}
+
+export function validateTemplateName(
+  validation: ICanaryFilterTemplateValidationMessages,
+  validationInput: ITemplateValidationInput,
+): ICanaryFilterTemplateValidationMessages {
+  const { editingTemplate, configTemplates } = validationInput;
+  const nextValidation = { ...validation };
```

Comment:
I don't totally see what the shallow copy is doing - why not a deep clone, why not use the same object reference?

Also, you could consider letting the caller pass the copy into the function on line 91 below, rather than requiring each validator to do this.